src/app/local-storage.service.ts

```diff
@@ -1,12 +1,26 @@
 import { Injectable } from '@angular/core';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class LocalStorageService {
   // https://developer.mozilla.org/es/docs/Web/API/Window/localStorage
-  constructor() { }
+
+  storage = window.localStorage;
+
+  constructor() {}
   getName() {
-    return 'LocalStorageService'
+    return 'LocalStorageService';
+  }
+
+  updateList(list) {
+    this.storage.setItem('list', JSON.stringify(list));
+    console.log(this.storage.getItem('list'));
+  }
+  getList() {
+    return JSON.parse(this.storage.getItem('list'));
```

Comment:
aca lo que tenes que tenes o podes hacer es hacer un 
JSON.parse(this.storage.getItem('list')).map(each => new TodoItem(each));

e implementar el constructor de TodoItem apra que tome los objetos, el problema es que typescript serializa y deseraliza los objetos sin tener en cuenta la clase, siempre es la clase object, por lo que si que si vos queres usar metodos de tuas clases vas a tener que instanciarlos manualmente

Reply:
tambien podes hacer lo que comenta tu compañero pero creo que es mas complicado, este es un tiuíco error al trabajar con serializacion de objetos en JS/TS